python/adjoint/utils.py

```diff
@@ -22,6 +22,11 @@
 # default finite difference step size when calculating Aᵤ
 FD_DEFAULT = 1e-3
 
+# Type hint for all components
+MEEP_COMPONENTS = Union[
+    mp.Ex, mp.Ey, mp.Ez, mp.Hx, mp.Hy, mp.Hz, mp.Dx, mp.Dy, mp.Dz, mp.Bx, mp.By, mp.Bz
```

Comment:
FYI- These are instances of integers, not types. I think the actual Python type for a field component in Meep is an int.

Reply:
Fixed, thanks @ianwilliamson!